src/commands/pins.rs

```diff
@@ -30,8 +30,8 @@ pub async fn list_pins(_sub_m: &clap::ArgMatches) -> Result<(), Box<dyn std::err
 /// * `_sub_m` - A reference to the ArgMatches struct containing parsed command-line arguments.
 ///              This is used to access the CID of the pin to retrieve and any additional options.
 pub async fn get_pin(_sub_m: &clap::ArgMatches) -> Result<(), Box<dyn std::error::Error>> {
-    let mut client = connect_to_gevulot(_sub_m).await?;
     if let Some(pin_cid) = _sub_m.get_one::<String>("cid") {
+        let mut client = connect_to_gevulot(_sub_m).await?;
```

Comment:
Instantiate client only after the CLI argument has been validated :slightly_smiling_face: